src/compiler/csharp_generator.cc

```diff
@@ -633,7 +627,7 @@ void GenerateClientStub(Printer* out, const ServiceDescriptor* service) {
             "methodfield", GetMethodFieldName(method));
         break;
       default:
-        GOOGLE_LOG(FATAL) << "Can't get here.";
+        break;
```

Comment:
I've noticed that currently csharp_generator is the only language that's using the GetMethodType() function -  other protoc plugins just use   method->client_streaming() and method->server_streaming() directly - perhaps this is the opportunity for simplification and unification. As a side effect we'd also get rid of the switch clause's `default:` branch
(same above)

See 
https://github.com/grpc/grpc/blob/86f93801e5d452cff40d6958d21a8eb211da0654/src/compiler/generator_helpers.h#L152

Reply:
Got it, updated the switch statements that use GetMethodType. Refactored the functions that expected a method type to query method directly